pinot-spi/src/main/java/org/apache/pinot/spi/exception/QException.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.exception;
+
+/**
+ * The base exception for query processing in Pinot.
+ *
+ * This exception is captured by the query engine and converted to a user-friendly message in a standardized format
+ * (ie converted into a JSON payload in an HTTP response).
+ *
+ * Contrary to {@code org.apache.pinot.common.response.ProcessingException}, which is usually designed to be thrown by
+ * {@code org.apache.pinot.common.exception.QueryException} in a non-allocating, non-throwing and stack-trace-less
+ * exception, this exception is designed to be thrown and caught by the query engine as any other standard Java
+ * exception.
+ */
+public class QException extends PinotRuntimeException {
```

Comment:
Suggest naming this `QueryException`. Since you are cleaning up `org.apache.pinot.common.exception.QueryException`, there is a TODO there to rename that to `QueryExceptionUtils`, and you can do it together to avoid name conflict

Reply:
I didn't want to have an even longer PR, but if you think it is the best way, I'll try to do that as well